addons/website_slides/models/slide_slide.py

```diff
@@ -745,6 +746,18 @@ def _parse_youtube_document(self, document_id, only_preview_fields):
             })
         return {'values': values}
 
+    def _extract_google_error_message(self, error):
+        try:
+            error = json.loads(error)
+            error = (error.get('error', {}).get('errors', []) or [{}])[0].get('reason')
+        except json.decoder.JSONDecodeError:
+            error = str(error)
+
+        if error == 'keyInvalid':
+            return _('Your Google API key is invalid, please update it into your settings.\nSettings > Website > Features > API Key')
+
+        return _('Could not fetch data from url. Document or access right not available:\n%s') % error
```

Comment:
```python
def _extract_google_error_message(self, result):
   """ Some docstring with code error structure example """
    try:
        error = json.loads(result).get('error', {}).get('errors', [{}])[0]
        reason = error.get('reason')
        if reason == 'keyInvalid':
            return _('Your Google API key is invalid, please update it into your settings.\nSettings > Website > Features > API Key')
    except json.decoder.JSONDecodeError:
        reason = str(result)

    return _('Could not fetch data from url. Document or access right not available:\n%s') % reason
```

Reply:
@dbeguin there's no error structure returned ;) it's just a string, with the "error reason" returned by Google (if the structure is know). Else, the structure is unknow (happen if we forgot a case)

Reply:
I was talking about Google response structure, not the one you return in the function.
Duh ! :smile: